sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/ChangeFeedPartitionReaderITest.scala

```diff
@@ -0,0 +1,315 @@
+  val cosmosEndpoint = TestConfigurations.HOST
+  val cosmosMasterKey = TestConfigurations.MASTER_KEY
+  val testId = UUID.randomUUID().toString
+  val sourceContainerResponse = cosmosClient.getDatabase(cosmosDatabase).createContainer(
+   "source_" + testId,
+   "/sequenceNumber",
+   ThroughputProperties.createManualThroughput(11000)).block()
+  val sourceContainer = cosmosClient.getDatabase(cosmosDatabase).getContainer(sourceContainerResponse.getProperties.getId)
+  val rid = sourceContainerResponse.getProperties.getResourceId
+  val continuationState = s"""{
+  "V": 1,
+  "Rid": "$rid",
+  "Mode": "INCREMENTAL",
+  "StartFrom": {
+    "Type": "BEGINNING"
+  },
+  "Continuation": {
+    "V": 1,
+    "Rid": "$rid",
+    "Continuation": [
+      {
+        "token": "1",
+        "range": {
+          "min": "",
+          "max": "FF"
+        }
+      }
+    ],
+    "Range": {
+      "min": "",
+      "max": "FF"
+    }
+  }
+}"""
+  val encoder = Base64.getEncoder
+  val encodedBytes = encoder.encode(continuationState.getBytes("UTF-8"))
+  val continuationStateEncoded = new String(encodedBytes, "UTF-8")
+
+  val changeFeedCfg = Map(
+   "spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+   "spark.cosmos.accountKey" -> cosmosMasterKey,
+   "spark.cosmos.database" -> cosmosDatabase,
+   "spark.cosmos.container" -> sourceContainer.getId(),
+   "spark.cosmos.read.inferSchema.enabled" -> "false",
+  )
+  var inputtedDocuments = 10
+  var lsn1 = 0L
+  var lsn2 = 0L
+  for (_ <- 0 until inputtedDocuments) {
+   lsn1 = ingestTestDocuments(sourceContainer, 1)
+   lsn2 = ingestTestDocuments(sourceContainer, 2)
+  }
+  inputtedDocuments *= 2
+
+  while (lsn1 != lsn2) {
+   if (lsn1 < lsn2) {
+    lsn1 = ingestTestDocuments(sourceContainer, 1)
+   } else {
+    lsn2 = ingestTestDocuments(sourceContainer, 2)
+   }
+   inputtedDocuments += 1
+  }
+
+  val structs = Array(
+   StructField("_rawBody", StringType, false),
+   StructField("_etag", StringType, false),
+   StructField("_ts", StringType, false),
+   StructField("id", StringType, false),
+   StructField("_lsn", StringType, false)
+  )
+  val schema = new StructType(structs)
+
+  val diagnosticContext = DiagnosticsContext(UUID.randomUUID(), "")
+  val cosmosClientStateHandles = initializeAndBroadcastCosmosClientStatesForContainer(changeFeedCfg)
+  val diagnosticsConfig = new DiagnosticsConfig(None, false, None)
+  val cosmosInputPartition = new CosmosInputPartition(NormalizedRange("", "FF"), Some(lsn1),
+   Some(continuationStateEncoded))
+  val changeFeedPartitionReader = new ChangeFeedPartitionReader(
+   cosmosInputPartition,
+   changeFeedCfg,
+   schema,
+   diagnosticContext,
+   cosmosClientStateHandles,
+   diagnosticsConfig,
+   ""
+  )
+  var count = 0
+
+  while (changeFeedPartitionReader.next()) {
+   changeFeedPartitionReader.get()
+   count += 1
+  }
+
+  count shouldEqual inputtedDocuments
+ }
+
+
+ "change feed partition reader" should "honor endLSN during split and should hang" in {
```

Comment:
should not hang?

Reply:
I am checking that it hangs if the endLSN has not been reached for 2 seconds then I start adding more documents and it will finish successfully. It should not complete on 304s because we know the endLSN exists when we begin, so it is verifying that behavior.